clarity/contracts/arkadiko-stacker-payer-v2-1.clar

```diff
@@ -0,0 +1,74 @@
+;; Implementation of Stacker Payer
+;; which allows users to redeem xSTX for STX
+
+(define-constant ERR-NOT-AUTHORIZED u22401)
+(define-constant ERR-EMERGENCY-SHUTDOWN-ACTIVATED u221)
+
+(define-data-var stacker-payer-shutdown-activated bool false)
+(define-data-var stx-redeemable uint u0)
+
+(define-read-only (get-stx-redeemable)
+  (var-get stx-redeemable)
+)
+
+(define-read-only (has-stx-redeemable)
+  (> (var-get stx-redeemable) u0)
+)
+
+(define-read-only (is-enabled)
+  (and
+    (not (unwrap-panic (contract-call? .arkadiko-dao get-emergency-shutdown-activated)))
+    (not (var-get stacker-payer-shutdown-activated))
+  )
+)
+
+(define-public (toggle-stacker-payer-shutdown)
+  (begin
+    (asserts! (is-eq tx-sender (contract-call? .arkadiko-dao get-guardian-address)) (err ERR-NOT-AUTHORIZED))
+
+    (ok (var-set stacker-payer-shutdown-activated (not (var-get stacker-payer-shutdown-activated))))
+  )
+)
+
+(define-public (return-stx-to-reserve (ustx-amount uint))
```

Comment:
Q: is this really needed or just in case something goes wrong?

Reply:
Just in case something goes wrong that no STX are lost.